File: 0485-max-consecutive-ones/0485-max-consecutive-ones.ts
function findMaxConsecutiveOnes(nums: number[]): number {
    //1 1 0 1 1 1
    let max:number=0;
    let ans:number=0;
    for(let num of nums){
        if(num===0) max=0;
        else max++;
        ans=Math.max(max, ans);


    }
    return ans;
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int cnt=0;
        int maxCnt=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==1){
                cnt++;
                maxCnt=max(maxCnt,cnt);
            }
            else{
                cnt=0;
            }
        }
        return maxCnt;
        
    }
};